independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Methods.java

```diff
@@ -408,4 +411,74 @@ public MethodVisitor visitMethod(int access, String name, String descriptor, Str
         }
     }
 
+    static class SubclassSkipPredicate implements Predicate<MethodInfo> {
+
+        private Set<MethodInfo> bridgeMethods = new HashSet<>();
+
+        @Override
+        public boolean test(MethodInfo method) {
+            if (isBridge(method)) {
+                bridgeMethods.add(method);
```

Comment:
I think we first need to build a list of all bridge methods before we move on to comparisons because a single class can contains both, bridge method and an abstract/concrete method that we need to process in defined order.